library/ccm.c

```diff
@@ -403,7 +404,8 @@ static const unsigned char res[NB_TESTS][32] = {
 int mbedtls_ccm_self_test( int verbose )
 {
     mbedtls_ccm_context ctx;
-    unsigned char out[32];
+    unsigned char plaintext[PLAINTEXT_MAX_LENGTH];
```

Comment:
I think you need a comment somewhere around here to state clearly that we're making the memory 'contigous and aligned' for the sake of some accelerators.